tests/test_search_readonly.py

```
@@ -0,0 +1,155 @@
+
+    def test_non_readonly_mode_still_links_file(self, tmp_path):
+        _qapp()
+        source = tmp_path / 'source' / 'drawing.pdf'
+        source.parent.mkdir(parents=True)
+        source.write_text('drawing contents')
+        bp_dir = tmp_path / 'blueprints'
+        bp_dir.mkdir()
+
+        # suppress_bp_link_notification=True avoids the modal QMessageBox
+        # confirmation on a successful link, keeping this test headless-safe.
+        app_context, show_error, save_settings = _make_app_context(
```

Ruff (0.16.1)

[warning] Unpacked variable `save_settings` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)